source/Xunit.Gherkin.Quick/CoreModel/Feature/Discovery/MissingFeatureDiscoveryModel.cs

```diff
@@ -23,10 +23,16 @@ internal sealed class MissingFeatureDiscoveryModel
             var referencedFiles = _featureClassInfoRepository.GetFeatureClassesInfo()
                 .Select(fc => fc.FeatureFilePath)
                 .ToList();
-
-            var newFiles = allFiles.Where(f => !referencedFiles.Contains(f))
+                
+            var referencedFilesByPattern = _featureClassInfoRepository.GetFeatureClassesInfo()
+                .SelectMany(fc => new FeatureFileRepository(fc.FileNameSearchPattern).GetFeatureFilePaths())

```

Comment:
Don't use repository like this, it will violate the single responsibility principle for the repository. Repository is only responsible to find all feature paths and get the feature by path. To find the paths that match the pattern, you need something else. You can create a helper class for that if you need to or write the code inside this method itself. But don't use repository.